pages/doing_api_first_development.md

```diff
@@ -11,96 +11,332 @@ sitemap:
-- You can use the specification file to mock your new server endpoints before they are released so you can more decouple frontend and backend development.
-- You don't need a live server to use your OpenAPI documentation.
+### The Pet Store v3
 
-### Using the OpenAPI-generator plugins
+Hereafter, this guide will be based on a [JHipster-ready version of Expanded Pet Store v3](api/api.yml) from the OpenAPI Initiative ([found here](https://github.com/OAI/OpenAPI-Specification/blob/main/examples/v3.0/petstore-expanded.yaml)). This JHipster-version sets the `server:url` to the JHipster defaults of `http://localhost:8081/api` (where the port defaults to `8081` and the generated controller has a base path of `/api`, matching the default `SecurityConfiguration#filterChain(HttpSecurity,  MvcRequestMatcher.Builder)` implementation).
+
+### Generating the Server Sources
 
-The OpenAPI specification file will be located at src/main/resources/swagger/api.yml and is used to generate endpoint interfaces that you can implement. 
-Those interfaces have default methods which answer with a `501 Not implemented` HTTP status and an empty body.
-Write your specification using a tool such as [swagger-editor](http://editor.swagger.io), put it in `src/main/resources/swagger/api.yml`, then run:
 ```bash
 ./mvnw generate-sources
 ```
 Or for gradle:
 ```bash
 ./gradlew openApiGenerate
 ```
-Then implement the "Delegate" interfaces generated in `${buildDirectory}/generated-sources/openapi/src/main/java/${package}/web/api/` with `@Service` classes.
 
-Example of code to write yourself for the famous [petstore](http://petstore.swagger.io):
+### Validate tooling
+
+Validate that the generated classes in `target/generated-sources` are on the classpath for implementation. Most IDEs look for and, automatically, configure these directories. Refer to the IDE's documentation to resolve any issues.
+
+### Review the Generated Classes
+
+Take a moment to review the `generated-sources`.
+
+```text
+target/generated-sources
+└── openapi
+    └── src
+        └── main
+            └── java
+                └── demo
+                    └── jhipster
+                        ├── service
+                        │   └── api
+                        │       └── dto
+                        │           ├── Error.java
+                        │           ├── NewPet.java
+                        │           └── Pet.java
+                        └── web
+                            └── api
+                                ├── ApiUtil.java
+                                ├── PetsApi.java
+                                ├── PetsApiController.java
+                                └── PetsApiDelegate.java
+```
+
+> ⚠️The OpenAPI Generator is responsible for the creation of DTOs ("models") and delegates. Implementations MUST not be placed in the `target/generated-sources`. These classes are subject to generation at anytime and SHOULD NOT be committed to source control.
```

Comment:
Confusing, `MUST not be place in target` and should not be comitted to source control. Do you mean `MUST be placed in target/generatedSources`?

Reply:
"MUST NOT"... Implementations should be put into the `src/main/java` (`project.build.sourceDirectory`) and the appropriate path to meet the technical structure tests. 

The `target` (`project.build.directory`) should be considered "ephemeral" and is deleted when running `mvn clean`. If the implementation code is put in the `target` directory, it has to be restored, i.e., `git co target/...` or something similar any time the _clean_ lifecycle is run. This might be often as the generator is bound to the `generate-sources` phase of the _default_ lifecycle and does not overwrite existing generated classes from the OpenAPI Specification ("swagger") document, `api.yml`.

Since implementation code would be part of the project, one would not want it deleted every time the plugin needs to update the generated source.